saleor/graphql/account/mutations/account.py

```diff
@@ -133,27 +132,29 @@ def clean_input(cls, info, instance, data, input_cls=None):
         return super().clean_input(info, instance, data, input_cls=None)
 
     @classmethod
-    @traced_atomic_transaction()
     def save(cls, info, user, cleaned_input):
-        password = cleaned_input["password"]
-        user.set_password(password)
-        user.search_document = search.prepare_user_search_document_value(
-            user, attach_addresses_data=False
-        )
-        if settings.ENABLE_ACCOUNT_CONFIRMATION_BY_EMAIL:
-            user.is_active = False
-            user.save()
-            notifications.send_account_confirmation(
-                user,
-                cleaned_input["redirect_url"],
-                info.context.plugins,
-                channel_slug=cleaned_input["channel"],
+        with traced_atomic_transaction():
```

Comment:
Would that work if we wrapped entire `perform_mutation` in the base mutation class, so that all functions actually happens in a transaction and we don't have to wrap these pieces individually?

Reply:
Keep in mind that `traced_atomic_transaction` creates `opentracing` block, that sends a span. Setting up this for each mutation will cause that we will gather there a huge amount of data.
 (Also not sure what about mutation like `tokenCreate`, that has user password as an input)

Reply:
I think that we should use transactions only when required. Having big transactions could generate problems, like that we miss some data when a transaction fails. A few months ago we speak about it with @patrys and decide it wasnt solve any of our problems.

Reply:
Just like @fowczarek mentioned - the whole purpose of this PR was to explicitly use transactions only when required.